File: 5/py/5.py
class Solution:
    def longestPalindrome(self, s: str) -> str:

        n = len(s)
        if n == 1 :
            return s
        
        maxS = ""
        curSOdd = ""
        curSEven = ""

        for i in range(n):
            curSOdd = self.centerExpend(i, i, s)

            if i < n-1 :
                curSEven = self.centerExpend(i, i+1, s)

            if len(maxS) < len(curSOdd) :
                maxS = curSOdd
            if len(maxS) < len(curSEven) :
                maxS = curSEven
            
        return maxS
        

    def centerExpend(self, l: int, r: int, s: str) -> str:
        subS = ""

        while l >= 0 and r < len(s) :
            if s[l] == s[r] :
                subS = s[l:r+1]
            else :
                break
            l -= 1
            r += 1
        
        return subS

File: 5/go/5.go
func longestPalindrome(s string) string {
    // check edge
    n := len(s)
    if n == 1 {
        return s
    }
    
    // use center expend strategy
    maxS := ""

    for i := 0; i < n; i++ {
        // find max
        curSOdd := ""
        curSEven := ""
        
        curSOdd = CenterExpend(i, i, s)

        if (i < n-1){
            curSEven = CenterExpend(i, i+1, s)
        }

        curMax := ""
        if len(curSOdd) > len(curSEven){
            curMax = curSOdd
        }else{
            curMax = curSEven
        }

        if len(curMax) > len(maxS) {
            maxS = curMax
        }
    }

    return maxS
}

func CenterExpend(rId int, lId int, s string) string {
    // check
    n := len(s)
    subS := ""

    for rId < n && lId >= 0 {
        if s[rId] == s[lId]{
            subS = s[lId:rId+1]
        }else{
            break
        }
        rId++
        lId--
    }

    return subS
}